annotation/tests/test_distribution.py

```diff
@@ -1277,3 +1291,238 @@ def test_get_page_number_combinations(
+            [{"user_id": "2"}],
+            [{"user_id": "0"}, {"user_id": "1"}],
+            [],
+        ),
+        (
+            "1",
+            ValidationSchema.hierarchical,
+            [{"user_id": "0"}, {"user_id": "1"}],
+            [{"user_id": "2"}],
+            [{"user_id": "1"}],
+            [{"user_id": "2"}],
+        ),
+    ),
+)
+def test_choose_validators_users(
+    split_multipage_doc_setup: str,
+    validation_type: ValidationSchema,
+    annotators: List[User],
+    validators: List[User],
+    annotation_tasks: List[Task],
+    expected_output: List[User],
+):
+    with patch(
+        "annotation.distribution.main.SPLIT_MULTIPAGE_DOC",
+        split_multipage_doc_setup,
+    ):
+        output = choose_validators_users(
+            validation_type, annotators, validators, annotation_tasks
+        )
+        assert output == expected_output
+
+
+@pytest.mark.parametrize(
+    ("users_id", "example_db", "expected_output"),
+    (
+        (
+            ["b908ccf0-6eab-4b8e-afb7-913372ddf4e5"],
+            ["b908ccf0-6eab-4b8e-afb7-913372ddf4e5"],
+            [User(user_id="b908ccf0-6eab-4b8e-afb7-913372ddf4e5")],
+        ),
+        (
+            [
+                "b908ccf0-6eab-4b8e-afb7-913372ddf4e5",
+                "00d37dc4-b7da-45e5-bcb5-c9b82965351c",
+            ],
+            ["b908ccf0-6eab-4b8e-afb7-913372ddf4e5"],
+            [
+                User(user_id="b908ccf0-6eab-4b8e-afb7-913372ddf4e5"),
+                User(user_id="00d37dc4-b7da-45e5-bcb5-c9b82965351c"),
+            ],
+        ),
+    ),
+)
+def test_prepare_users(
+    users_id: List[UUID], example_db: Tuple[UUID], expected_output: List[User]
+):
+    db = Mock()
+
+    def read_user_side_effect(db, user_id):
+        print("READ ME")
+        if user_id in example_db:
+            return User(user_id=user_id)
+        return None
+
+    def create_user_side_effect(db, user_id):
+        print("CREATE ME")
+        return User(user_id=user_id)
+
+    with patch(
+        "annotation.distribution.main.read_user",
+        side_effect=read_user_side_effect,
+    ), patch(
+        "annotation.distribution.main.create_user",
+        side_effect=create_user_side_effect,
+    ):
+        output = prepare_users(db, users_id)
+        for a, e in zip(output, expected_output):
+            assert a.user_id == e.user_id
+
+
+@pytest.mark.parametrize(
+    (
+        "users",
+        "all_job_pages_sum",
+        "average_job_pages",
+        "users_default_load",
+        "users_overall_load",
+        "expected_output",
+        "expected_share_loads",
+    ),
+    (
+        (
+            [{"user_id": "0", "overall_load": 10, "default_load": 10}],
+            10,  # all_job_pages_sum
+            10.0,  # average_job_pages
+            10,  # users_default_load
+            10,  # users_overall_load
+            1.0,  # expected_output
+            [1],  # expected_share_loads
```

Comment:
you have one parametrized case, no need to use it then

Reply:
To make sure all parts of code in the function are tested, I'll add one more test case